envi/archs/arm/const.py

```diff
@@ -3,6 +3,32 @@
 MODE_JAZELLE    = 2
 MODE_THUMBEE    = 3
 
+'''
+bitfields to support different versions
+For simplicity will use 0xFFFF to say all.
+Note that thumb and arm are both represented
+thumbee is shown but may never be fully implimented
+since was already depreciated
+'''
+archBitMask = [
```

Comment:
please make these "constants":

REV_ARMv4 = 0b.....
REV_ARMv5 = ...
REV_ARMv5T = ...

then group them altogether in a dictionary:
ARCH_REVS = { num : name for name,num in globals().items() if name.startswith('REV_') }

an interactive example:
In [1]: REV_FOO = 1

In [2]: ARCH_REVS = { num : name for name,num in globals().items() if name.startswith('REV_') }

In [3]: ARCH_REVS
Out[3]: {1: 'REV_FOO'}


you don't want to be wielding tuples when actually using the thing.  you want to use the constant, and let it be "the thing"  like:

if self.capabilities & ARVv5T:
    dostuff()